modules/fbs-core/api/src/main/scala/de/thm/ii/fbs/controller/SubmissionController.scala

```diff
@@ -250,4 +261,84 @@ class SubmissionController {
       throw new ForbiddenException()
     }
   }
+
+  @GetMapping(value = Array("/{uid}/courses/{cid}/tasks/{tid}/submissions/{sid}/content"))
+  @ResponseBody
+  def getContent(@PathVariable uid: Int, @PathVariable cid: Int, @PathVariable tid: Int, @PathVariable sid: Int,
+                 req: HttpServletRequest, res: HttpServletResponse): ResponseEntity[InputStreamResource] = {
+    val user = authService.authorize(req, res)
+    val task = taskService.getOne(tid).get
+
+    val privileged = user.id == uid || user.hasRole(GlobalRole.ADMIN, GlobalRole.MODERATOR)
+
+    if (privileged) {
+      submissionService.getOne(sid, uid) match {
+        case Some(submission) => {
+          val file: File = storageService.getFileSolutionFile(submission)
+          ResponseEntity.ok()
+            .contentType(MediaType.APPLICATION_OCTET_STREAM)
+            .contentLength(file.length())
+            .header("Content-Disposition", s"attachment;filename=task_${task.id}.fbs-submission")
+            .body(new InputStreamResource(Files.newInputStream(file.toPath, StandardOpenOption.DELETE_ON_CLOSE)))
+        }
+        case None => throw new ResourceNotFoundException()
+      }
+    } else {
+      throw new ResourceNotFoundException()
+    }
+  }
+
+  @GetMapping(value = Array("/{uid}/courses/{cid}/tasks/submissions/content"))
+  @ResponseBody
+  def solutionsOfCourse(@PathVariable uid: Int, @PathVariable cid: Int,
+                        req: HttpServletRequest, res: HttpServletResponse): ResponseEntity[InputStreamResource] = {
+    val user = authService.authorize(req, res)
+
+    val privileged = user.id == uid || user.hasRole(GlobalRole.ADMIN, GlobalRole.MODERATOR)
+
+    if (privileged) {
+      val submissionList = submissionService.getLatestSubmissionByCourse(cid)
+      val usersList: ListBuffer[List[User]] = ListBuffer()
+      val t = submissionList.map(s => s.taskID).distinct
+      val listSubInDir: ListBuffer[List[File]] = ListBuffer()
+      t.foreach(taskid => {
+        val tmp = submissionList.filter(s => s.taskID == taskid)
+        listSubInDir += tmp.map(submission => storageService.getFileSolutionFile(submission))
+        usersList += tmp.map(submission => userService.find(submission.userID.get).get)
+      })
+      val f = new File("tmp")
+      Archiver.packSubmissionsInDir(f, listSubInDir, usersList, t)
+      ResponseEntity.ok()
+        .contentType(MediaType.APPLICATION_OCTET_STREAM)
+        .contentLength(f.length())
+        .header("Content-Disposition", s"attachment;filename=course_$cid.fbs-submission")
+        .body(new InputStreamResource(Files.newInputStream(f.toPath, StandardOpenOption.DELETE_ON_CLOSE)))
+    } else {
+      throw new ResourceNotFoundException()
+    }
+  }
+
+  @GetMapping(value = Array("/{uid}/courses/{cid}/tasks/{tid}/submissions/content/download"))
```

Comment:
- `/{uid}` makes no sense here because you are querying the data of all users.
- `/download` should be removed, otherwise it is inconsistent